shiny/ui/_include_helpers.py

```diff
@@ -0,0 +1,247 @@
+from __future__ import annotations
+
+__all__ = ("include_js", "include_css")
+
+import glob
+import hashlib
+import os
+import shutil
+import tempfile
+
+# TODO: maybe these include_*() functions should actually live in htmltools?
+from htmltools import HTMLDependency, Tag, TagAttrValue, tags
+
+from .._docstring import add_example
+from .._typing_extensions import Literal
+
+# TODO: it's bummer that, when method="link_files" and path is in the same directory
+# as the app, the app's source will be included. Should we just not copy .py/.r files?
+
+
+@add_example()
+def include_js(
+    path: str,
+    *,
+    method: Literal["link", "link_files", "inline"] = "link",
+    **kwargs: TagAttrValue,
+) -> Tag:
+    """
+    Include a JavaScript file
+
+    Parameters
+    ----------
+    path
+        A path to a JS file.
+    method
+        One of the following:
+          * ``"link"``: Link to the JS file via a :func:`~ui.tags.script` tag. This
+            method is generally preferrable to ``"inline"`` since it allows the browser
+            to cache the file.
+          * ``"link_files"``: Same as ``"link"``, but also allow for the CSS file to
+            request other files within ``path``'s immediate parent directory (e.g.,
+            ``fetch()`` the contents of another file). Note that this isn't the default
+            behavior because you should **be careful not to include files in the same
+            directory as ``path`` that contain sensitive information**. A good general
+            rule of thumb to follow is to have ``path`` be located in a subdirectory of
+            the app directory. For example, if the app's source is located at
+            ``/app/app.py``, then ``path`` should be somewhere like
+            ``/app/js/custom.js`` (and all the other relevant accompanying 'safe' files
+            should be located under ``/app/js/``).
+          * ``"inline"``: Inline the JS file contents within a :func:`~ui.tags.script`
+            tag.
+    *kwargs
+        Attributes which are passed on to `~ui.tags.script`
+
+
+    Returns
+    -------
+    A :func:`~ui.tags.script` tag.
+
+    Note
+    ----
+    This places a :func:`~ui.tags.script` tag in the :func:`~ui.tags.body` of the
+    document. If instead, you want to place the tag in the :func:`~ui.tags.head` of the
+    document, you can wrap it in ``head_content`` (in this case, just make sure you're
+    aware that the DOM probably won't be ready when the script is executed).
+
+    .. code-block:: python
+
+
+
+    ui.fluidPage(
+        head_content(ui.include_js("custom.js")),
+    )
+
+    # Alternately you can inline Javscript by changing the method.
+    ui.fluidPage(
+        head_content(ui.include_js("custom.js", method = "inline")),
+    )
+
+    See Also
+    --------
+    ~ui.tags.script
+    ~include_css
+    """
+
+    if method == "inline":
+        return tags.script(read_utf8(path), type="text/javascript", **kwargs)
```

Comment:
The `type="text/javascript"` here will conflict with `type="module"`, if the user provides the latter argument.

We can just remove the `type="text/javascript"` here. According to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type):
> Authors are encouraged to omit the attribute if the script refers to JavaScript code rather than specify a MIME type.